### PR TITLE
Dapp Account Selection & Defaults

### PR DESCRIPTION
**Pull request by jacogr**
Closes https://github.com/ethcore/parity/issues/4328

1. Move Dapp Permissions dialog to Portal
2. Extend selection to include default account
3. Extend store to read & write default accounts

Implementation notes -

1. Removed animations from the Portal - could be re-added using either tween.js (react-portal samples) or ReactCSSTransitionGroup (FB preferred) - however gave up on both of those, nice-to-have an spending an inordinate amount of time on it.
2. Setting defaults to be done now in Signer overlay as well (https://github.com/ethcore/parity/issues/4368)

Sample -

parity 2017-01-31 12-37-18

### CODE REVIEW

**Comment by ngotchac** [comment]
Might want to apply this in order for all accounts on one line to have the same height

```diff
diff --git a/js/src/modals/DappPermissions/dappPermissions.css b/js/src/modals/DappPermissions/dappPermissions.css
index 28fd9cd..8b7c03a 100644
--- a/js/src/modals/DappPermissions/dappPermissions.css
+++ b/js/src/modals/DappPermissions/dappPermissions.css
@@ -21,10 +21,12 @@
 
 .container {
   margin-top: 1.5em;
-  overflow-y: scroll;
+  overflow-y: auto;
 }
 
 .item {
+  display: flex;
+  flex: 1;
   position: relative;
 
   .overlay {
diff --git a/js/src/ui/SectionList/sectionList.css b/js/src/ui/SectionList/sectionList.css
index 4fa1bbf..0cb6dd4 100644
--- a/js/src/ui/SectionList/sectionList.css
+++ b/js/src/ui/SectionList/sectionList.css
@@ -39,8 +39,8 @@
     .item {
       box-sizing: border-box;
       cursor: pointer;
+      display: flex;
       flex: 0 1 33.33%;
-      height: 100%;
       opacity: 0.75;
       padding: 0.25em;
       transition: all 0.75s cubic-bezier(0.23, 1, 0.32, 1);
```

File: js/src/modals/DappPermissions/dappPermissions.css [hunk 1]
```diff
@@ -15,33 +15,58 @@
 /* along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+.modal {
+  flex-direction: column;
+}
+
+.container {
+  margin-top: 1.5em;
+  overflow-y: scroll;
```
Comment (ngotchac) [inline js/src/modals/DappPermissions/dappPermissions.css]: Change to `auto` so we don't have unnecessary scroll bars

**Comment by ngotchac** [comment]
Only the scroll bar is a blocking issue. For the _all accounts don't have the same height on one line_ issue, could go either way. However it would feel more consistent with what we have currently.

**Comment by jacogr** [comment]
@ngotchac I actually prefer to solve that issue another way, i.e. the AccountCard should always return the same height. The issue is with applying it, you now have rows that look uneven. Very distracting.

(With my flow-hover-pretty branch, that is exactly the route I decided to go - make sure everything always looks consistent as to not have these funnies.)

For now your solution is probably a bit better than what is these, so will apply as-is. (Will hoever log the above to look into)

**Pull request merged**